components/storage/client_storage.rs

```diff
@@ -32,7 +40,9 @@ impl ClientStorageThreadFactory for GenericSender<ClientStorageThreadMessage> {
 pub struct ClientStorageThread {
     _base_dir: PathBuf,
     _generic_sender: GenericSender<ClientStorageThreadMessage>,
-    generic_receiver: GenericReceiver<ClientStorageThreadMessage>,
+    routed_receiver: RoutedReceiver<ClientStorageThreadMessage>,
+    storage_key_connections: HashMap<StorageKeyConnectionId, StorageKeyConnection>,
```

Comment:
So the point of having more then just the origin is defined at https://privacycg.github.io/storage-partitioning/

And I wonder how `StorageKeyConnectionId` would meet the goal of partinioning storage per top-level site. `StorageKeyConnectionId` seems both too fine-grained, and not related to what top-level site the storage request relates. 

So whatever key scheme we adopt, it should be chosen in the light of storage-partitioning, otherwise just the origin would do.